core/src/main/java/org/springframework/security/authorization/AuthorityAuthorizationManager.java

```diff
@@ -124,7 +124,7 @@ private AuthorityAuthorizationManager(String... authorities) {
 	@Override
 	public AuthorizationDecision check(Supplier<Authentication> authentication, T object) {
 		boolean granted = isGranted(authentication.get());
-		return new AuthorizationDecision(granted);
+		return new AuthorityAuthorizationDecision(granted, this.authorities);
```

Comment:
Given my other comment about changing this constructor to take a collection of `GrantedAuthority`s, I think that this class should store a collection of `GrantedAuthority`s.